### PR TITLE
test(replays): add test for getCurrentUrl feature

### LINKED ISSUE #34923: Add test for getCurrentUrl feature
### Problem Statement

Add test for replays/getCurrentUrl() feature

### PR DESCRIPTION
**Pull request by SDaian**
### Description

- Add a test for getCurrentUrl feature.
- Add a mock file

Closes: #34923 

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: tests/js/spec/utils/replays/getCurrentUrl.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    createEventTransaction(),
+    createRrWebEvents(),
+    createReplayEvents()
+  );
+}
+
+describe('getCurrentUrl', () => {
+  it('should return the current url given a replay', () => {
+    const replay = createReplay() as ReplayReader;
+    expect(replay).toBeDefined();
```
Comment (billyvg) [inline tests/js/spec/utils/replays/getCurrentUrl.spec.tsx:23]: This isn't really doing much, it's not testing anything from `getCurrentUrl()`

File: tests/js/spec/utils/replays/getCurrentUrl.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    expect(getCurrentUrl(replay, 0)).toBe('https://sourcemaps.io/');
+  });
+
+  it('should return a different current url if Navigation action happens', () => {
```
Comment (billyvg) [inline tests/js/spec/utils/replays/getCurrentUrl.spec.tsx:27]: This doesn't really reflect what we're testing, there isn't a "navigation action" happening in the test

File: tests/js/spec/utils/replays/getCurrentUrl.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import getCurrentUrl from 'sentry/utils/replays/getCurrentUrl';
+import ReplayReader from 'sentry/utils/replays/replayReader';
+
+import {
+  createEventTransaction,
+  createReplayEvents,
+  createRrWebEvents,
+} from './mocks/events.fixture';
+
+const CURRENT_TIME_MS = 55114;
```
Comment (benvinegar) [inline tests/js/spec/utils/replays/getCurrentUrl.spec.tsx:10]: Given this is a very specific time (and not necessarily "current" time"), I would just declare it in the test function that uses it, and give it an explicit name, e.g.:

```
const timeAfterNavigationEventOccurs = 55114;
```

File: tests/js/spec/utils/replays/getCurrentUrl.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (billyvg) [inline tests/js/spec/utils/replays/getCurrentUrl.spec.tsx:1]: Is this needed?

**Review by billyvg — COMMENTED** [review]
This is a good start, but there's a few things missing that would make it a solid set of tests.

1) We have so much "realistic" mock data, that it's not immediately obvious what we're testing. But more importantly is...
2) We're only testing a very simple test case when there's only 1 navigation breadcrumb.

We're not testing what happens if:
* currentTime is < the first navigation event
* currentTime is between two navigation events
* currentTime is exactly in the middle of two navigation events
* we have 2 navigation events with exactly the same time
* what happens if currentTime is > all other events
...etc

good tests will try to cover for all scenarios that *can* happen.

File: tests/js/spec/utils/replays/getCurrentUrl.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+import ReplayReader from 'sentry/utils/replays/replayReader';
+
+import {
+  createEventTransaction,
+  createReplayEvents,
+  createRrWebEvents,
+} from './mocks/events.fixture';
+
+function createReplay(): ReplayReader | null {
+  return ReplayReader.factory(
```
Comment (billyvg) [inline tests/js/spec/utils/replays/getCurrentUrl.spec.tsx:11]: `factory()` needs to be updated, it takes a single object argument now.

**Pull request closed without merging**